server/fishtest/templates/tests_view.mak

```diff
@@ -514,10 +518,41 @@
   document
     .getElementById("moon")
     .addEventListener("click", () => setHighlightTheme("dark"));
-</script>
 
-<script>
-  document.title = "${page_title} | Stockfish Testing";
+  async function downloadPGNs() {
+    try {
+      const response = await fetch(`/api/run_pgns/${run["_id"]}.pgn.gz`);
+      if (!response.ok) {
+        if (response.status === 404) {
+          alertError("No games found for this run");
+        } else {
+          alertError("Unable to download PGNs");
+        }
+        return;
+      }
+      const blob = await response.blob();
+
+      // Check if the blob is empty
+      if (blob.size === 0) {
+        alertError("Blob is empty!");
+        return;
+      }
+      
+      const url = window.URL.createObjectURL(blob);
+      const a = document.createElement("a");
+      a.href = url;
+      a.download = `${run["_id"]}.pgn.gz`;
+      document.body.append(a);
+      a.click();
+      document.body.removeChild(a);
+    } catch (error) {
+      alertError("Network error: please check your network!");
+    }
+  }
+
+  document
+    .getElementById("download_games")?
+    .addEventListener("click", downloadPGNs);
```

Comment:
I copied the whole JS cose in a file to catch a couple of errors shown by the browser
```js
prettier --write aaaaa.js
aaaaa.js
[error] aaaaa.js: SyntaxError: Unexpected token (89:5)
[error]   87 |   document
[error]   88 |     .getElementById("download_games")?
[error] > 89 |     .addEventListener("click", downloadPGNs);
[error]      |     ^
[error]   90 |
[error]   91 |   let fetchedTasksBefore = false;
[error]   92 |   async function handleRenderTasks(){
```

prettier does not break that line
```js
document.getElementById("download_games")?.addEventListener("click", downloadPGNs);
```

I don't know the fix for this one:
```js
prettier --write aaaaa.js
aaaaa.js
[error] aaaaa.js: SyntaxError: Unexpected token, expected ")" (96:11)
[error]   94 |       await toggleTasks();
[error]   95 |     })
[error] > 96 |      if (${str(tasks_shown).lower()})
[error]      |           ^
[error]   97 |        await renderTasks();
[error]   98 |   }
[error]   99 |
```

Reply:
the other one is not an error, it's only a valid syntax inside the mako, the code inside `${}` is a python code.

BTW my prettier here does break it.. I think 80 characters is default for you no?
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATCYCuBbBMB0A5nDAKIA2cusAQgJ4CSqAFADpoQDuU5EAhqgD6hPrgDO7AJQB+fANSkAbngAyASzHwocAE5tw5NWADW7ADQACdN14CACgHEAcmMkBuEGZAQADjDXQYsigfDo6XHahCEEofOScfHRBXgBGOnwmJADKonDq2sgAZnFicKnpmTBZPhlqUITIMDqYZSCl2GqNza1wAB4+umrUMHEAKrpQoWpwMcXkpV5idYSUAIqYEPBFJa0AVmK9WctrG1tIcwsgAI6ncHbhPjEgfGIAtNpwqJ+eIE18aoZ6gBhCDYbB8ZDPcjkH5LeqUACCMCaahSmHgdl0+Tg23mrQAFjBsOQAOr4tTwMQ1MBwLLRClqRQUuiQsBiZIgRQtBhQL6wLJgHRqPwI3lZGB0Si4y4+cKlEnpHyQ2UzXTKH51Uo6GD3PiEcHS1o1HRayEpPgpOAwryyuowElqVAwfHIAAcAAYvDo4Dc1N7dfqIecdl4RikHU6XUgAExeTClUYW2YhkBUS2oL6oFR8eqYPVwABiEB04ORy0hfHREBAAF8a0A

Reply:
I got the errors in the browser, then I copied the code to find the issue with the formatting.
The first point was an issue with the spaces/tabs perhaps.
The error message in the browser for the second point is disappeared now in the browser.